src/main/java/maksym/fedorenko/bookstore/model/OrderItem.java

```diff
@@ -0,0 +1,49 @@
+package maksym.fedorenko.bookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Table(name = "order_items")
+@SQLDelete(sql = "UPDATE order_items SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted=false")
+@Data
+public class OrderItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "order_id")
+    private Order order;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "book_id")
+    private Book book;
```

Comment:
does it make sense to make this mapping bidirectional? like with cart items

motivation - if a book gets deleted for example - then cart items are deleted as well. but what about order items?) same regarding update

and now we're back to the soft delete problem. book supports soft delete, as well as order and order item. when a book is deleted - you will end up with some deleted order items. however these order items weren't deleted because of order check out. so what to do during audit? how to differentiate between two types of deleted order items?

Reply:
For now, I don't fully understand your comment and how to solve this problem in general, so I'll leave it. And then I think I’ll try fix it